gym/vector/vector_env.py

```diff
@@ -49,129 +48,112 @@ def __init__(self, num_envs, observation_space, action_space):
     ):
+        """Reset the vector environments asynchronously."""
         pass
 
     def reset_wait(
         self,
-        seed: Optional[Union[int, List[int]]] = None,
+        seed: Optional[Union[int, list[int]]] = None,
         return_info: bool = False,
         options: Optional[dict] = None,
     ):
+        """Resets the vector environments while waiting for the results."""
         raise NotImplementedError()
 
     def reset(
         self,
         *,
-        seed: Optional[Union[int, List[int]]] = None,
+        seed: Optional[Union[int, list[int]]] = None,
         return_info: bool = False,
         options: Optional[dict] = None,
     ):
-        r"""Reset all parallel environments and return a batch of initial observations.
+        """Reset all parallel environments and return a batch of initial observations.
+
+        Args:
+            seed: The environment reset seeds
+            return_info: If to return the info
+            options: If to return the options
 
-        Returns
-        -------
-        observations : element of :attr:`observation_space`
+        Returns:
             A batch of observations from the vectorized environment.
         """
         self.reset_async(seed=seed, return_info=return_info, options=options)
         return self.reset_wait(seed=seed, return_info=return_info, options=options)
 
     def step_async(self, actions):
+        """Steps through the environment action asynchronously."""
         pass
 
     def step_wait(self, **kwargs):
+        """Steps through the vector environment while waiting for results."""
         raise NotImplementedError()
 
     def step(self, actions):
-        r"""Take an action for each parallel environment.
-
-        Parameters
-        ----------
-        actions : element of :attr:`action_space`
-            Batch of actions.
-
-        Returns
-        -------
-        observations : element of :attr:`observation_space`
-            A batch of observations from the vectorized environment.
+        """Take an action for each parallel environment.
 
-        rewards : :obj:`np.ndarray`, dtype :obj:`np.float_`
-            A vector of rewards from the vectorized environment.
+        Args:
+            actions: element of :attr:`action_space` Batch of actions.
 
-        dones : :obj:`np.ndarray`, dtype :obj:`np.bool_`
-            A vector whose entries indicate whether the episode has ended.
-
-        infos : list of dict
-            A list of auxiliary diagnostic information dicts from each parallel environment.
+        Returns:
+            Batch of observations, rewards, done and infos
         """
-
         self.step_async(actions)
         return self.step_wait()
 
     def call_async(self, name, *args, **kwargs):
+        """Calls a method name for each parallel environment asynchronously."""
         pass
 
     def call_wait(self, **kwargs):
+        """Calls a method and wait for the results."""
```

Comment:
I thin the call is actually being done from `call_async`.

Reply:
Updated to "After calling a method in :meth:`call_async`, this function collects the results."